benchmarks/Microsoft.AspNetCore.Routing.Performance/LinkGeneration/SingleRouteRouteValuesBasedEndpointFinderBenchmark.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Routing.LinkGeneration
+{
+    public class SingleRouteRouteValuesBasedEndpointFinderBenchmark : LinkGenerationBenchmarkBase
+    {
+        private IEndpointFinder<RouteValuesAddress> _finder;
+        private TestEndpointFinder _baseFinder;
+        private RequestContext _requestContext;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            var template = "Products/Details";
+            var defaults = new { controller = "Products", action = "Details" };
+            var requiredValues = new { controller = "Products", action = "Details" };
+
+            var endpoint = CreateEndpoint(template, defaults, requiredValues: requiredValues, routeName: "ProductDetails");
+            var services = CreateServices();
+            _finder = services.GetRequiredService<IEndpointFinder<RouteValuesAddress>>();
+            _baseFinder = new TestEndpointFinder(endpoint);
+
+            _requestContext = CreateCurrentRequestContext();
+        }
+
+        [Benchmark(Baseline = true)]
+        public void Baseline()
+        {
+            var actual = _baseFinder.FindEndpoints(address: 0);
+            GC.KeepAlive(actual);
```

Comment:
What is the purpose of the KeepAlive? Either remove or add a comment.

Repeat for other benchmarks in this file.